lib/Backends/CPU/CPUDeviceManager.h

```diff
@@ -41,7 +41,11 @@ class CPUDeviceManager : public QueueBackedDeviceManager {
 
 public:
   CPUDeviceManager(const DeviceConfig &config, size_t maxMemory = 2000000000)
```

Comment:
config already has memory param. Let's remove additional maxMemory here and make sure that `deviceMemory` is set properly in the config (make sure callers set it properly when needed, do not pass it as a separate param).

For the case of Habana where GlowHabanaMemory is defined from the GFLAGS or llvm params we should use it as long as it's set otherwise fallback to what config says.

Reply:
> config already has memory param. Let's remove additional maxMemory here and make sure that `deviceMemory` is set properly in the config (make sure callers set it properly when needed, do not pass it as a separate param).
> 
> For the case of Habana where GlowHabanaMemory is defined from the GFLAGS or llvm params we should use it as long as it's set otherwise fallback to what config says.

OK. I created a new deviceConfig inside the `createCPUDeviceManager` or `createInterpreterDeviceManager` if it is set by command line. It seems that since it is a const that I could not set it directly. To keep the change smaller, I just created a new deviceConfig.